metrics-servlets/src/main/java/com/codahale/metrics/servlets/HealthCheckServlet.java

```diff
@@ -86,7 +88,13 @@ public void init(ServletConfig config) throws ServletException {
     @Override
     protected void doGet(HttpServletRequest req,
                          HttpServletResponse resp) throws ServletException, IOException {
-        final SortedMap<String, HealthCheck.Result> results = runHealthChecks();
+        final Map<String, HealthCheck.Result> results;
+        String name = getHealthCheckName(req);
+        if (name != null) {
+            results = Collections.singletonMap(name, registry.runHealthCheck(name));
```

Comment:
Actually that method throws NoSuchElementException when check is not found.
You should create method that would check if that check exists and only then call method.

If check is not existing you should return empty map so responses (all/one) would be consistent.